var/spack/repos/builtin/packages/megadock/package.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Megadock(MakefilePackage, CudaPackage):
+    """an ultra-high-performance protein-protein docking for
+       heterogeneous supercomputers"""
+
+    homepage = "http://www.bi.cs.titech.ac.jp/megadock/"
+    url      = "http://www.bi.cs.titech.ac.jp/megadock/archives/megadock-4.0.3.tgz"
+
+    version('4.0.3', sha256='c1409a411555f4f7b4eeeda81caf622d8a28259a599ea1d2181069c55f257664')
+
+    variant('mpi', description='Enable MPI', default=False)
+
+    depends_on('fftw')
+    depends_on('mpi', when='+mpi')
+
+    def edit(self, spec, prefix):
+        # point cuda samples relative to cuda installation
+        filter_file('/opt/cuda/6.5/samples', '$(CUDA_INSTALL_PATH)/samples',
+                    'Makefile', string=True)
+
+        # need to link calcrg to compiler's math impl
+        # libm seems to be present in most compilers
+        mathlib = '-lm'
+
+        # prefer libimf with intel
+        if '%intel' in spec:
+            mathlib = '-limf'
```

Comment:
FYI, the few lines above can be written:
```python
mathlib = '-limf' if '%intel' in spec else '-lm'
```
to make the package more compact / readable.